Objects/moduleobject.c

```diff
@@ -333,6 +348,27 @@ PyModule_FromDefAndSpec2(PyModuleDef* def, PyObject *spec, int module_api_versio
         goto error;
     }
 
+#ifdef Py_GIL_DISABLED
+    PyThreadState *tstate = _PyThreadState_GET();
+    const PyConfig *config = _PyInterpreterState_GetConfig(interp);
+    if (gil_slot == Py_MOD_GIL_USED && config->enable_gil == _PyConfig_GIL_DEFAULT) {
+        if (_PyEval_EnableGIL(tstate)) {
```

Comment:
FWIW, this is not where I would look for code that turns the GIL back on.  I'd look in Python/import.c (or maybe Python/importdl.c).  That said, I'm not sure how that code could know whether or not the GIL should be enabled for a module.  It would have basically have to repeat the code you have here or we'd store the result on the module (or module def) and refer to that.

I'm not sure that's worth it and yet calling `_PyEval_EnableGIL()` here still doesn't seem right.

Reply:
I'm fine with leaving it like you have it, but would you mind leaving a comment noting that ideally we'd find a way to do this where it more naturally fits (in import.c or importdl.c)?